libvast/src/detail/posix.cpp

```diff
@@ -60,6 +60,85 @@ int uds_accept(int socket) {
   return fd;
 }
 
+uds_datagram_sender::uds_datagram_sender(uds_datagram_sender&& other) noexcept
+  : src_fd{other.src_fd}, dst{other.dst} {
+  // Invalidate the original copy to avoid closing when `other` is destroyed.
+  other.src_fd = -1;
+}
+
+uds_datagram_sender&
+uds_datagram_sender::operator=(uds_datagram_sender&& other) noexcept {
+  src_fd = other.src_fd;
+  // Invalidate the original copy to avoid closing when `other` is destroyed.
+  other.src_fd = -1;
+  dst = other.dst;
+  return *this;
+}
+
+uds_datagram_sender::~uds_datagram_sender() {
+  if (src_fd != -1)
+    ::close(src_fd);
+}
+
+caf::expected<uds_datagram_sender>
+uds_datagram_sender::make(const std::string& path) {
+  auto result = uds_datagram_sender{};
+  result.src_fd = ::socket(AF_UNIX, SOCK_DGRAM, 0);
+  if (result.src_fd < 0)
+    return caf::make_error(
+      ec::system_error,
+      "failed to obtain an AF_UNIX DGRAM socket: ", ::strerror(errno));
+  // Create a unique temporary directory for a place to bind the sending side
+  // to. There is no mktemp variant for sockets, so that is unfortunately
+  // necessary.
+  // NOTE: The temporary directory will be removed at the end of this function.
+  char mkd_template[] = u8"/tmp/vast-XXXXXX\0socket";
```

Comment:
Why the UTF-8 string literal? Can you add a comment for the future reader?

Reply:
That is a leftover that I forgot to remove.